### PR TITLE
OpenLineage: Add parent_run_id for COMPLETE and FAIL events

### PR DESCRIPTION
**Pull request by kacpermuda**
Add information about parent_run_id to FAIL and COMPLETE event in Openlineage. 

This way You can easily assign all the events to their parent run.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by JDarDagran — APPROVED** [review]

File: tests/providers/openlineage/plugins/test_openlineage_adapter.py [hunk 1]
```diff
@@ -240,6 +389,63 @@ def test_emit_failed_event(mock_stats_incr, mock_stats_timer):
...
+    adapter.fail_task(
+        run_id=run_id,
+        end_time=event_time,
+        parent_job_name="parent_job_name",
+        parent_run_id="parent_run_id",
+        job_name="job",
+        task=OperatorLineage(
+            inputs=[Dataset(namespace="bigquery", name="a.b.c"), Dataset(namespace="bigquery", name="x.y.z")],
+            outputs=[Dataset(namespace="gs://bucket", name="exported_folder")],
+            run_facets={"sql1": SqlJobFacet(query="SELECT 1;")},
```
Comment (mobuchowski) [inline tests/providers/openlineage/plugins/test_openlineage_adapter.py]: Why do you attach `JobFacet` to run facets?

File: tests/providers/openlineage/plugins/test_openlineage_adapter.py [hunk 2]
```diff
@@ -206,6 +296,63 @@ def test_emit_complete_event(mock_stats_incr, mock_stats_timer):
...
+                    facets={
+                        "parent": ParentRunFacet(
+                            run={"runId": "parent_run_id"},
+                            job={"namespace": "default", "name": "parent_job_name"},
+                        ),
+                        "parentRun": ParentRunFacet(
+                            run={"runId": "parent_run_id"},
+                            job={"namespace": "default", "name": "parent_job_name"},
+                        ),
+                        "sql1": SqlJobFacet(query="SELECT 1;"),
```
Comment (mobuchowski) [inline tests/providers/openlineage/plugins/test_openlineage_adapter.py]: The same question 🙂

File: tests/providers/openlineage/plugins/test_openlineage_adapter.py (hunk 1 above)
Comment (kacpermuda) [reply tests/providers/openlineage/plugins/test_openlineage_adapter.py]: I wanted to attach any sample Facet that will not interfere with automatically added facets. I will change that to use Run typ facet.

File: tests/providers/openlineage/plugins/test_openlineage_adapter.py (hunk 2 above)
Comment (kacpermuda) [reply tests/providers/openlineage/plugins/test_openlineage_adapter.py]: Let's use the above topic to discuss this, as this is the same case.

File: tests/providers/openlineage/plugins/test_openlineage_adapter.py (hunk 1 above)
Comment (kacpermuda) [reply tests/providers/openlineage/plugins/test_openlineage_adapter.py]: Updated.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**